### PR TITLE
Prepare unreal_helpers crate for publishing

### PR DESCRIPTION
**Pull request by localcc**

### CODE REVIEW

File: unreal_helpers/README.md [hunk 1]
```diff
@@ -1,3 +1,18 @@
...
+
+## Documentation
+
+Crate documentation is published to
+[docs.rs/unreal_helpers](https://docs.rs/unreal_helpers/).
+
+## Usage
+
+The crate can be added to a Rust project as a dependency by running the command
+`cargo add gvas`.
```
Comment (konsti219) [inline unreal_helpers/README.md:13]: ?

Comment (localcc) [reply unreal_helpers/README.md:13]: guh

File: unreal_asset/src/reader/asset_reader.rs [hunk 2]
```diff
@@ -32,7 +32,7 @@ pub trait AssetReader: AssetTrait {
     fn read_i64<T: ByteOrder>(&mut self) -> io::Result<i64>;
     fn read_f32<T: ByteOrder>(&mut self) -> io::Result<f32>;
     fn read_f64<T: ByteOrder>(&mut self) -> io::Result<f64>;
-    fn read_fstring(&mut self) -> io::Result<Option<String>>;
+    fn read_fstring(&mut self) -> Result<Option<String>, Error>;
     fn read_exact(&mut self, buf: &mut [u8]) -> io::Result<()>;
-    fn read_bool(&mut self) -> io::Result<bool>;
+    fn read_bool(&mut self) -> Result<bool, Error>;
```
Comment (konsti219) [inline unreal_asset/src/reader/asset_reader.rs:37]: change here too?

File: unreal_asset/src/reader/raw_reader.rs [hunk 3]
```diff
@@ -188,15 +188,15 @@ impl AssetReader for RawReader {
...
+        Ok(self.cursor.read_fstring()?)
     }
 
     fn read_exact(&mut self, buf: &mut [u8]) -> io::Result<()> {
         self.cursor.read_exact(buf)
     }
 
-    fn read_bool(&mut self) -> io::Result<bool> {
+    fn read_bool(&mut self) -> Result<bool, Error> {
         Ok(self.read_u8()? != 0)
```
Comment (konsti219) [inline unreal_asset/src/reader/raw_reader.rs:200]: why this way?

File: unreal_asset/src/reader/asset_writer.rs [hunk 4]
```diff
@@ -21,7 +21,7 @@ pub trait AssetWriter: AssetTrait {
     fn write_i64<T: ByteOrder>(&mut self, value: i64) -> io::Result<()>;
     fn write_f32<T: ByteOrder>(&mut self, value: f32) -> io::Result<()>;
     fn write_f64<T: ByteOrder>(&mut self, value: f64) -> io::Result<()>;
-    fn write_fstring(&mut self, value: Option<&str>) -> io::Result<usize>;
+    fn write_fstring(&mut self, value: Option<&str>) -> Result<usize, Error>;
     fn write_all(&mut self, buf: &[u8]) -> io::Result<()>;
-    fn write_bool(&mut self, value: bool) -> io::Result<()>;
+    fn write_bool(&mut self, value: bool) -> Result<(), Error>;
```
Comment (konsti219) [inline unreal_asset/src/reader/asset_writer.rs:26]: here too?

**Review by konsti219 — CHANGES_REQUESTED** [review]

**Review by konsti219 — APPROVED** [review]

**Pull request merged**